### DISCUSSION TITLE
AssertionError: some folds are missing

### DISCUSSION THREAD

**Discussion by mchlsdrv** [MIC-DKFZ/nnUNet#507] [Q&A] [answered]
Hi, I've successfully completed all 5 folds (0-4), as required, and have all the directories present, though when I ran the `nnUNet_find_best_configuration -m 3d_fullres -t 101--strict` I get the "AssertionError: some folds are missing
" error.

  File "/usr/local/lib/python3.6/dist-packages/nnunet/postprocessing/consolidate_postprocessing.py", line 29, in collect_cv_niftis
    assert all([isdir(i) for i in folders_folds]), "some folds are missing"
AssertionError: some folds are missing

What could be the reason for that?

Thanks in advance'
Michael

**Comment by FabianIsensee** [comment, marked as answer]
You need to run all 5 folds of the cross validation. 
Best,
Fabian

**Reply by mchlsdrv** [reply]
I ran nnUNet_train 3d_fullres nnUNetTrainerV3 101 0, 1, 2, 3, 4

isn't it the 5-fold CV?

**Reply by FabianIsensee** [reply]
It is but maybe some of your folds did not complete yet? Check your filesystem and verify that they all ran, including the validation step (validation_raw subfolder with summary.json in it)

**Reply by Peaceandmaths** [reply]
Is it possible to run the find_best_configuration on only a certain number of folds and not all 5 ? I also had the issue of stopping the training and re-running so not all of them are complete. I'm just experimenting, I don't actually need to train all 5 for now, just want to see the whole pipeline on a small dataset.

**Comment by abergsneider** [comment]
Correct me if I'm wrong. But on the command you mentioned to have run: `nnUNet_find_best_configuration -m 3d_fullres -t 101--strict`, you're only specifying one configuration.

In order to run this command, you need to **specify >1 configurations** with 5-folds each. @sidoroffmichael

**Reply by FabianIsensee** [reply]
You can run this with only one configuration. It will always return that configuration as the best, so its rather pointless. But it should run nonetheless

**Reply by mchlsdrv** [reply]
Thanks @abergsneider , @FabianIsensee. 

@FabianIsensee - if i re-run on of the folds, should I run all of them? I mean, I ran folds 0, 1, 2, then I accidentally ran fold 1 again, then I stopped it and ran fold 3 and 4. Maybe that causing the problem? 
Thanks in advance.

**Reply by FabianIsensee** [reply]
please rerun the validations of all folds:
```
nnUNet_train 3d_fullres nnUNetTrainerV2 101 0 -val
nnUNet_train 3d_fullres nnUNetTrainerV2 101 1 -val
nnUNet_train 3d_fullres nnUNetTrainerV2 101 2 -val
nnUNet_train 3d_fullres nnUNetTrainerV2 101 3 -val
nnUNet_train 3d_fullres nnUNetTrainerV2 101 4 -val
```
And verify that this gives no error

**Reply by mchlsdrv** [reply]
I'm re-running all the training now. I'll see if it fixes the problem and notify you here. If the problem remains - I'll try to rerun the validation. Thanks for your help.
